clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -218,4 +242,57 @@ public boolean seekToNewSource(long l) throws IOException {
             return false;
         }
     }
+
+    private Location pathToLocation(Path path) {
+        if (!path.isAbsolute()) {
+            path = new Path(this.workingDirectory, path);
+        }
+
+        URI uri = path.toUri();
+        if (uri.getScheme() != null && uri.getPath().isEmpty()) {
+            return null;
+        }
+
+        Location loc = new Location();
+        loc.setRepository(uri.getHost());
+        String s = uri.getPath().substring(1);
+        int i = s.indexOf("/");
+        if (i == -1) {
+            loc.setRef(s);
+        } else {
+            loc.setRef(s.substring(0, i));
+            loc.setPath(s.substring(i+1));
+        }
+        return loc;
+    }
+
+    private static class Location {
+        public String getRepository() {
+            return repository;
+        }
+
+        public void setRepository(String repository) {
+            this.repository = repository;
+        }
+
+        public String getRef() {
+            return ref;
+        }
+
+        public void setRef(String ref) {
+            this.ref = ref;
+        }
+
+        public String getPath() {
+            return path;
+        }
+
+        public void setPath(String path) {
+            this.path = path;
+        }
+
+        private String repository;
+        private String ref;
```

Comment:
To implement the delete operation I'm going to need a branch. I saw that ref could be either a commit id or a branch. In the context of a file system, can't we always expect to get a branch? if so, it will clearer calling it a branch